### PR TITLE
Postponed switch to run cmd changes

### PR DESCRIPTION
**Pull request by kevinclark19a**
## Description

As a follow up to PR #2050, this PR adds bad the changes that were postponed for the DCR fix.

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and Travis.CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azurelinuxagent/common/osutil/default.py [hunk 1]
```diff
@@ -630,9 +635,20 @@ def umount_dvd(self, chk_err=True, mount_point=None):
...
-        retcode = shellutil.run("eject {0}".format(dvd))
-        if chk_err and retcode != 0:
-            raise OSUtilError("Failed to eject dvd: ret={0}".format(retcode))
+        try:
+            shellutil.run_command(["eject", dvd])
+        except shellutil.CommandError as cmd_err:
+            if chk_err:
+                
+                msg = """Failed to eject dvd: ret={0}
+                [stdout]
```
Comment (narrieta) [inline azurelinuxagent/common/osutil/default.py:644]: need to remove extra spaces from message

File: azurelinuxagent/common/osutil/gaia.py [hunk 2]
```diff
@@ -150,11 +150,11 @@ def deploy_ssh_pubkey(self, username, pubkey):
     def eject_dvd(self, chk_err=True):
         logger.warn('eject is not supported on GAiA')
 
-    def mount(self, device, mount_point, option="", chk_err=True):
-        logger.info('mount {0} {1} {2}', device, mount_point, option)
-        if 'udf,iso9660' in option:
+    def mount(self, device, mount_point, option=[], chk_err=True):
+        logger.info('mount {0} {1} {2}', device, mount_point, " ".join(option))
+        if filter(option, lambda opt: 'udf,iso9660' in opt):
```
Comment (narrieta) [inline azurelinuxagent/common/osutil/gaia.py:155]: the original condition `'udf,iso9660' in option` should also work with the array of strings

File: azurelinuxagent/common/osutil/gaia.py [hunk 3]
```diff
@@ -150,11 +150,11 @@ def deploy_ssh_pubkey(self, username, pubkey):
...
             ret, out = super(GaiaOSUtil, self).mount(
-                device, mount_point, option=option.replace('udf,iso9660', 'udf'),
+                device, mount_point, option=list(map(option, lambda opt: opt.replace('udf,iso9660', 'udf'))),
```
Comment (narrieta) [inline azurelinuxagent/common/osutil/gaia.py]: I get this error creating the map:

```
>>> option = ["hello", "goodbye", 'udf,iso9660']
>>> map(option, lambda opt: opt.replace('udf,iso9660', 'udf'))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'function' object is not iterable
```

Comment (narrieta) [reply azurelinuxagent/common/osutil/gaia.py]: Maybe
```
>>> ['udf' if opt == 'udf,iso9660' else opt for opt in option]
['hello', 'goodbye', 'udf']
>>> 
```
?

**Review by pgombar — COMMENTED** [review]
LGTM except the extra whitespace around the error message

File: azurelinuxagent/common/osutil/gaia.py (hunk 2 above)
Comment (kevinclark19a) [reply azurelinuxagent/common/osutil/gaia.py]: I wanted to preserve the original functionality. The original condition would only work in the case that string `'udf,iso9660'` is exactly in the array, whereas the current implementation is fuzzier (like the original condition would've been with the original `option`).

Comment (narrieta) [reply azurelinuxagent/common/osutil/gaia.py]: ok, let's leave as is to preserve the previous behavior

**Review by narrieta — APPROVED** [review]

File: azurelinuxagent/common/osutil/gaia.py [hunk 4]
```diff
@@ -151,8 +151,6 @@ def eject_dvd(self, chk_err=True):
         logger.warn('eject is not supported on GAiA')
 
     def mount(self, device, mount_point, option=None, chk_err=True):
-        logger.info('mount {0} {1} {2}', device, mount_point, " ".join(option))
```
Comment (pgombar) [inline azurelinuxagent/common/osutil/gaia.py]: Why was this removed?

**Review by larohra — APPROVED** [review]

File: azurelinuxagent/common/osutil/gaia.py (hunk 4 above)
Comment (kevinclark19a) [reply azurelinuxagent/common/osutil/gaia.py]: I had added this logging statement in the other run_command PR, and noticed that it should be removed in this one.

Comment (pgombar) [reply azurelinuxagent/common/osutil/gaia.py]: Thanks, I didn't realize you had introduced it, I thought it was there from the beginning. Makes sense then!
